### ISSUE TITLE
Confusing Demotion Behavior

### ISSUE THREAD

**Issue by OkJaybird** [habitat-sh/habitat#6303]
Demoting a package from a channel still leaves that version running for any services subscribing to updates from that channel. This is really confusing from a management perspective, because running services can drift from a channel's latest version. Imagine the following scenario:

1. Package ver 1.2.3 is currently in the stable channel.
2. I am running that package as a service subscribed to the stable channel. It picks up and is currently running ver 1.2.3.
3. I promote package ver 4.5.6 to the stable channel.
4. My running service now picks up and is running package ver 4.5.6.
5. I demote version 4.5.6 from stable. Now the latest version in stable is 1.2.3 again.
6. My running service remains at ver 4.5.6.

My expected behavior is to have my service reflect the latest version available in the channel it's subscribed to (`hab svc load --channel my_channel okjaybird/my_service`). In the above scenario, the service running would revert to version 1.2.3 after the demotion occurred.

Basically, I would expect to see parity between what's in the channel and what's running for services subscribed to that channel, but this is not currently the case. I had some good discussion with @ericcalabretta about this, and thought we should open a ticket.

**Comment by chefsalim** [comment]
@OkJaybird thanks for opening the issue. The current promote/demote behavior at the supervisor end is definitely not symmetric, and we can use this thread to discuss what it would take to make it so. 

From a functionality and expectation standpoint, I agree a symmetric demote that does a rollback would be quite useful, and is probably the expectation of the large majority of hab users.

I'll refrain from commenting on the design/implementation changes required, however at a minimum it would need to be the following four steps (each fairly small):
1. Detect the version downgrade for the service
2. Stop the service
3. Uninstall the package (and possibly remove from cache)
4. Restart the service

Note that we do not guarantee the reliability/safety aspect of actually doing a demote (or a promote for that matter), as that is up to how the service is developed. But having the level of control in operability seems like a significant value add.

**Comment by ericcalabretta** [comment]
We can close this issue, The behavior was added as part of the `--update-condition` flag. With `track-channel` set this promote/demote behavior is present.

**Comment by mwrock** [comment]
closing according to note above
